pkgs/development/tools/analysis/qcachegrind/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchurl, bash, cmake, qt57, graphviz,  qmakeHook, makeQtWrapper, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "kcachegrind-${version}";
+  version = "16.12.2";
+
+  src = fetchurl {
+    url = "http://download.kde.org/stable/applications/${version}/src/${name}.tar.xz";
+    sha256 = "0wcizav1z3n31n7qcc05mx42vl2rv71c7haaa8vn0ma39rj4iaqv";
+  };
+
+  buildInputs = [ qt57.qtbase graphviz ];
+  nativeBuildInputs = [ qmakeHook makeQtWrapper makeWrapper ];
+
+  postInstall = stdenv.lib.optionalString (stdenv.isDarwin) ''
+     mkdir -p $out/bin
+     cp -r converters/dprof2calltree $out/bin/dprof2calltree
+     cp -r converters/hotshot2calltree.cmake $out/bin/hotshot2calltree
+     cp -r converters/memprof2calltree $out/bin/memprof2calltree
+     cp -r converters/op2calltree $out/bin/op2calltree
+     cp -r converters/pprof2calltree $out/bin/pprof2calltree
+     chmod -R +x $out/bin/
```

Comment:
To include converters, you should list their dependencies in [build inputs](https://github.com/NixOS/nixpkgs/blob/18684a4892bc202590a51a8bee251dc1b4fcf3b7/pkgs/applications/kde/kcachegrind.nix#L15). (Users may disable them with e.g. `qcachegrind.override { php = null; }`.)